servercom/servercom.go

```diff
@@ -0,0 +1,138 @@
+package serverCom
+
+import (
+	"fmt"
+	mm "github.com/mattermost/platform/model"
+)
+/*
+ServerCom acts as a mitigator between the frontend and the mattermost model API.
+*/
+type ServerCom struct {
+	Client mm.Client
+	Team mm.Team
+	Channel *mm.Channel
+}
+
+/*
+Startup accepts the url and login credentials for a user, and returns a new serverCom struct.
+*/
+func Startup(url string, username string, password string) ServerCom {
```

Comment:
That makes sense. I had some of the earlier methods return errors initially. I changed them to print instead when I noticed the setters didn't actually return any value, and i thought it might be weird to have something like `err :=  MethodCall()`. But it makes sense having the freedom to chose how to handle them.